### DISCUSSION TITLE
TypeScript: Controller defaultValue incorrect type

### DISCUSSION THREAD

**Discussion by revich2** [react-hook-form/react-hook-form#5465] [Q&A] [answered]
**Describe the bug**
When you use Controller for field with name containing array literal, then defaultValue can't catch type from form values types.

**To Reproduce**
  - Open Codesandbox link
  - See the typescript error on field defaultValue - Type 'string' is not assignable to type 'undefined'.ts(2322)

**Codesandbox link (Required)**
https://codesandbox.io/s/lively-haze-zbqzs?file=/src/App.tsx

**Expected behavior**
Prop defaultValue should have the right type from form values types

**Comment by bluebill1049** [comment]
working as expected for me: https://codesandbox.io/s/hardcore-jepsen-x1ey7?file=/src/App.tsx

**Comment by thanh-nguyen-95** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/44762180/120583907-4002f080-c459-11eb-82d8-9be80638c944.png)

Removing the first error where you type `Controller` solves the problem. You don't need to type `Controller`, type will be inferred through `control` and `name`.

```tsx
<Controller
  name={`a.c.${1}.d` as const}
  control={props.control}
  defaultValue={"text"}
  render={({ field }) => <span>{field.value}</span>}
/>
```
